File: 1850-minimum-length-of-string-after-deleting-similar-ends/minimum-length-of-string-after-deleting-similar-ends.py
class Solution:
    def minimumLength(self, s: str) -> int:
        i = 0
        j = len(s)-1
        if s[i]!=s[j]:return len(s)
        c=s[i]
        while(i<j):
            while(i<j-1 and s[i] == c):i =i+1
            while(j>i-1 and s[j] == c):j =j-1
            if(s[i] == s[j]):c=s[i]
            else:break
        return j-i+1

File: 1850-minimum-length-of-string-after-deleting-similar-ends/minimum-length-of-string-after-deleting-similar-ends.cpp
class Solution {
public:
    int minimumLength(string s) {
        
        int n=s.size();
        int i=0;
        int j=n-1;
       if(s[i]!=s[j])return n;
       char c = s[i];
        while(i<j){
             
            while(s[i]==c and i<j-1 ){
                i++;
            }
            while(j>i-1 and s[j]==c){
                j--;
            }
            if(s[i]==s[j]){
                c=s[i];
            }
            else 
            {   
                break;
            }
        }
        return j-i+1;
    }
};